apps/mobile/src/screens/chat/components/categories-form.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useFormController,
+	useState,
+} from "~/libs/hooks/hooks";
+import {
+	DEFAULT_ERROR_MESSAGE,
+	toastMessage,
+} from "~/libs/packages/toast-message/toast-message";
+import { globalStyles } from "~/libs/styles/styles";
+import { quizCategoriesValidationSchema } from "~/packages/quiz/quiz";
+import { actions as quizActions } from "~/slices/quiz/quiz";
+
+const CategoriesForm: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const { categories } = useAppSelector((state) => state.quiz);
+
+	const ZERO_CATEGORIES = 0;
+	const hasCategories = Boolean(categories.length > ZERO_CATEGORIES);
+
+	useEffect(() => {
+		if (!hasCategories) {
+			void dispatch(quizActions.getQuizCategories());
+		}
+	}, [dispatch, hasCategories]);
+
+	const [submittedCategories, setSubmittedCategories] = useState<null | string>(
```

Comment:
As I understand this useState is a temporary solution, so let's add TODO comment to remove it

Reply:
I think such an approach may contradict [B34] quality criteria

Reply:
That's the kind of comment you can leave.
We will remove it as soon as we fix implementation of onSubmit.

Reply:
Ok

Reply:
I don't think we need to have a temporary solution at this point. Why not take values from the useAppForm?

Reply:
<img width="576" alt="Знімок екрана 2024-09-03 о 17 03 23" src="https://github.com/user-attachments/assets/276c239f-6814-484f-86a4-55595693603e">

I suggest to get rid of the text string

Reply:
All logic that is not related to final logic should be marked with a TODO comment or removed, because you may forget and leave some unnecessary changes

Reply:
I completely agree with you and have a counterproposal to add this rule to the quality-criteria so that these changes can be implemented throughout the entire project.

Reply:
Please don't mark as resolved unfixed comments! The same goes for the Constructor type!

Reply:
As you wish, but it'll take much more time for code review process than if you fix everything at once